### PR TITLE
Misc cleanup of DSG

### PR DESCRIPTION
**Pull request by andrewballantyne**
Resolves: #480

### CODE REVIEW

File: frontend/src/SDKInitialize.tsx [hunk 1]
```diff
@@ -36,7 +37,17 @@ const SDKInitialize: React.FC = ({ children }) => {
...
 
-  return <AppInitSDK configurations={config}>{ready ? children : null}</AppInitSDK>;
+  return (
+    <AppInitSDK configurations={config}>
+      {ready ? (
+        children
+      ) : (
+        <Bullseye>
+          <Spinner />
+        </Bullseye>
```
Comment (andrewballantyne) [inline frontend/src/SDKInitialize.tsx:47]: Juntao noted this -- we should always show spinners now... this was really old code so we had not made that change yet and I forgot to review it again once I started rebasing.

File: frontend/src/app/NavSidebar.tsx [hunk 2]
```diff
@@ -48,9 +48,9 @@ const NavSidebar: React.FC = () => {
       <NavList>
         {userNavData.map((item) =>
           isNavDataGroup(item) ? (
-            <NavGroup item={item} pathname={routerLocation.pathname} />
+            <NavGroup key={item.id} item={item} pathname={routerLocation.pathname} />
           ) : (
-            <NavHref item={item} pathname={routerLocation.pathname} />
+            <NavHref key={item.id} item={item} pathname={routerLocation.pathname} />
```
Comment (andrewballantyne) [inline frontend/src/app/NavSidebar.tsx:53]: Oops -- saw the error when I was testing

File: frontend/src/services/roleBindingService.ts [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
+    queryOptions: { name: rbName, ns: projectName },
   });
 };
 
 export const createRoleBinding = (data: RoleBinding): Promise<RoleBinding> => {
-  const url = `/api/rolebindings`;
-  return axios.post(url, data).then((response) => {
-    return response.data;
-  });
+  return k8sCreateResource({ model: RoleBindingModel, resource: data });
```
Comment (andrewballantyne) [inline frontend/src/services/roleBindingService.ts:13]: The API calls change the way we get data -- with the SDK vs what we did in our custom API. Needed to convert this to help with the utility of verify resource.

**Review by lucferbux — APPROVED** [review]
/lgtm

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Review by dlabaj — APPROVED** [review]
/lgtm

**Pull request merged**